### ISSUE TITLE
Bad EXE format when installing vcrun6sp6

### ISSUE THREAD

**Issue by maluhia** [Winetricks/winetricks#1154]
Trying to run a program with wine, but I get `cannot import dll :isskin.dll`. Thus, I am instructed to install `vcrun6sp6` with winetricks.

I attempted with both a 64bit and 32bit prefix, but the installation fails. Output is below. Note that both corefonts and vcrun2008 installed successfully with winetricks. Output of error follows:

```
user@debian:~/.cache/winetricks$ WINEPREFIX=~/.wine32 winetricks vcrun6sp6
Using winetricks 20180815 - sha256sum: 29e664ec1e879271a33167bf367deaa188cbe803d44b540e692bd669c44a2804 with wine-4.0-rc1 (Staging) and WINEARCH=win32
Executing w_do_call vcrun6sp6
Executing load_vcrun6sp6 
Executing mkdir -p /home/user/.cache/winetricks/vcrun6sp6
Executing cd /home/user/.cache/winetricks/vcrun6sp6
Downloading https://download.microsoft.com/download/1/9/f/19fe4660-5792-4683-99e0-8d48c22eed74/Vs6sp6.exe to /home/user/.cache/winetricks/vcrun6sp6
--2018-12-09 12:40:25--  https://download.microsoft.com/download/1/9/f/19fe4660-5792-4683-99e0-8d48c22eed74/Vs6sp6.exe
Resolving download.microsoft.com (download.microsoft.com)... 23.36.176.218, 2600:1409:12:2af::e59, 2600:1409:12:285::e59
Connecting to download.microsoft.com (download.microsoft.com)|23.36.176.218|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 63799912 (61M) [application/octet-stream]
Saving to: ‘Vs6sp6.exe’

Vs6sp6.exe           100%[===================>]  60.84M  1.53MB/s    in 42s     

2018-12-09 12:41:13 (1.44 MB/s) - ‘Vs6sp6.exe’ saved [63799912/63799912]

Executing cd /home/user/.cache/winetricks
Executing cabextract -q /home/user/.cache/winetricks/vcrun6sp6/Vs6sp6.exe -d /home/user/.wine32/dosdevices/c:/windows/temp/_vcrun6sp6 -F vcredist.exe
Executing cd /home/user/.wine32/dosdevices/c:/windows/temp/_vcrun6sp6
Executing rm -f /home/user/.wine32/dosdevices/c:/windows/system32/comcat.dll
Executing rm -f /home/user/.wine32/dosdevices/c:/windows/system32/msvcrt.dll
Executing rm -f /home/user/.wine32/dosdevices/c:/windows/system32/oleaut32.dll
Executing rm -f /home/user/.wine32/dosdevices/c:/windows/system32/olepro32.dll
Executing rm -f /home/user/.wine32/dosdevices/c:/windows/system32/stdole2.tlb
000b:fixme:winediag:start_process Wine Staging 4.0-rc1 is a testing version containing experimental patches.
000b:fixme:winediag:start_process Please mention your exact version when filing bug reports on winehq.org.
wine: Bad EXE format for C:\windows\temp\_vcrun6sp6\vcredist.exe.
------------------------------------------------------
vcrun6sp6 installation failed
------------------------------------------------------
```
Versions:
wine-4.0-rc1 (Staging)
winetricks: 20180815 - sha256sum: 29e664ec1e879271a33167bf367deaa188cbe803d44b540e692bd669c44a2804

**Comment by austin987** [comment]
It's a bug in wine-staging, works fine in plain wine, 32 and 64-bit.

**Comment by austin987** [comment]
(tested using 4.0-rc1)

**Comment by maluhia** [comment]
@austin987 Thank you for the quick reply.

I removed wine-staging and installed wine from the debian repos. (version 3.03-2). I still receive the error with 64bit and 32bit staging. I created a fresh prefix just to make sure:
```
WINEPREFIX=~/.wine32 WINEARCH="win32" winecfg
wine: created the configuration directory '/home/user/.wine32'
0012:err:ole:marshal_object couldn't get IPSFactory buffer for interface {00000131-0000-0000-c000-000000000046}
0012:err:ole:marshal_object couldn't get IPSFactory buffer for interface {6d5140c1-7436-11ce-8034-00aa006009fa}
0012:err:ole:StdMarshalImpl_MarshalInterface Failed to create ifstub, hres=0x80004002
0012:err:ole:CoMarshalInterface Failed to marshal the interface {6d5140c1-7436-11ce-8034-00aa006009fa}, 80004002
0012:err:ole:get_local_server_stream Failed: 80004002
0014:err:ole:marshal_object couldn't get IPSFactory buffer for interface {00000131-0000-0000-c000-000000000046}
0014:err:ole:marshal_object couldn't get IPSFactory buffer for interface {6d5140c1-7436-11ce-8034-00aa006009fa}
0014:err:ole:StdMarshalImpl_MarshalInterface Failed to create ifstub, hres=0x80004002
0014:err:ole:CoMarshalInterface Failed to marshal the interface {6d5140c1-7436-11ce-8034-00aa006009fa}, 80004002
0014:err:ole:get_local_server_stream Failed: 80004002
0019:err:mscoree:LoadLibraryShim error reading registry key for installroot
0019:err:mscoree:LoadLibraryShim error reading registry key for installroot
0019:err:mscoree:LoadLibraryShim error reading registry key for installroot
0019:err:mscoree:LoadLibraryShim error reading registry key for installroot
0010:err:winediag:SECUR32_initNTLMSP ntlm_auth was not found or is outdated. Make sure that ntlm_auth >= 3.0.25 is in your path. Usually, you can find it in the winbind package of your distribution.
wine: configuration in '/home/user/.wine32' has been updated.

user@debian:~$ WINEPREFIX=~/.wine32 winetricks vcrun6sp6
Using winetricks 20180815 - sha256sum: 29e664ec1e879271a33167bf367deaa188cbe803d44b540e692bd669c44a2804 with wine-3.0.3 (Debian 3.0.3-2) and WINEARCH=win32
Executing w_do_call vcrun6sp6
Executing load_vcrun6sp6 
Executing cabextract -q /home/user/.cache/winetricks/vcrun6sp6/Vs6sp6.exe -d /home/user/.wine32/dosdevices/c:/windows/temp/_vcrun6sp6 -F vcredist.exe
Executing cd /home/user/.wine32/dosdevices/c:/windows/temp/_vcrun6sp6
Executing rm -f /home/user/.wine32/dosdevices/c:/windows/system32/comcat.dll
Executing rm -f /home/user/.wine32/dosdevices/c:/windows/system32/msvcrt.dll
Executing rm -f /home/user/.wine32/dosdevices/c:/windows/system32/oleaut32.dll
Executing rm -f /home/user/.wine32/dosdevices/c:/windows/system32/olepro32.dll
Executing rm -f /home/user/.wine32/dosdevices/c:/windows/system32/stdole2.tlb
wine: Bad EXE format for C:\windows\temp\_vcrun6sp6\vcredist.exe.
------------------------------------------------------
vcrun6sp6 installation failed
------------------------------------------------------
```

**Comment by austin987** [comment]
Actually, this is likely cabextract related. What version do you have?

See https://github.com/kyz/libmspack/issues/22 / https://github.com/Winetricks/winetricks/issues/1120 for more info

**Comment by maluhia** [comment]
@austin987 I see. I have cabextract 1.9

**Comment by austin987** [comment]
You'll need to upgrade/downgrade to a working version. See:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=914263
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=912687

**Comment by maluhia** [comment]
Sounds like it was just fixed today and will be pushed to cabextract 1.9.2
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=914263#62

Thanks again
